File: Device Screen/Display Metrics/MainActivity.java
package com.example.myapplication;

import android.content.Context;
import android.os.Bundle;
import android.util.DisplayMetrics;
import android.view.WindowManager;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        onDisplayMetrics();
    }

    private void onDisplayMetrics() {
        DisplayMetrics displayMetrics = getResources().getDisplayMetrics();
        WindowManager windowManager = (WindowManager) getSystemService(Context.WINDOW_SERVICE);
        windowManager.getDefaultDisplay().getMetrics(displayMetrics);
        int widthPixels = displayMetrics.widthPixels;
        int heightPixels = displayMetrics.heightPixels;
        float density = displayMetrics.density;
        int densityDpi = displayMetrics.densityDpi;
        float scaledDensity = displayMetrics.scaledDensity;
        float xdpi = displayMetrics.xdpi;
        float ydpi = displayMetrics.ydpi;
    }

}

File: Kotlin/Functions/Device Screen/Display Metrics/MainActivity.kt
package kz.snation.myapplication

import android.content.Context
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.DisplayMetrics
import android.view.WindowManager

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        onDisplayMetrics()
    }

   private fun onDisplayMetrics() {
       val displayMetrics: DisplayMetrics = resources.displayMetrics
       val windowManager: WindowManager = getSystemService(Context.WINDOW_SERVICE) as WindowManager
       windowManager.defaultDisplay.getMetrics(displayMetrics)
       val widthPixels = displayMetrics.widthPixels
       val heightPixels = displayMetrics.heightPixels
       val density = displayMetrics.density
       val densityDpi = displayMetrics.densityDpi
       val scaledDensity = displayMetrics.scaledDensity
       val xdpi = displayMetrics.xdpi
       val ydpi = displayMetrics.ydpi
   }
}
